docs/design/storage-node-deployment-windows-home.md

```diff
@@ -0,0 +1,165 @@
+* Delete really old stuff.
+* Ensure we limit the size of logs...
+
+### Resource Limits
+
+Ensure we:
+* can set limits to memory usage,
+* can set limits to CPU usage.
+* send graceful shutdown message and restart storage node if memory usage is too high.
+* Windows specific apis also exist for limiting process memory and cpu usage.
+* limit CPU might be to limit number of cores it runs on in go.
+
+## Testing
+
+Verify program can be run with windows defender firewall and at least one other 3rd party firewall.
+
+## Design Overview
+
+### Automatic Updates and Handling failures
+
+General
+* Windows firewall and other 3rd party firewalls can block storage node operations.
+    * [isportallowed](https://docs.microsoft.com/en-us/windows/win32/api/netfw/nf-netfw-inetfwmgr-isportallowed) windows api function can be used to make sure we are allowed through firewall
+    * Can we add code to detect if firewall is blocking storage node operations?
+    * Unblock storage node operator in firewall settings.
+    * Can we detect if windows firewall is running?
+
+When downloading a new binary:
+* download fails
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* out-of-space for downloading
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* filesystem read-only
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* Man in the Middle attacks/corrupted binary
+    * Log error in log file if possible
+    * Verify binary hashes with message from version server and with output of a hashing algorithm (shasum256)
+    * Retry download on next cycle that checks if storage node is up to date.
+
+When swapping in a new binary:
+* computer crashes during swapping
+    * Log error in log file if possible
+    * automatic updater checks binary version and reruns download/swap steps.
+* deletion/stopping of the old binary fails.
+    * Log error in log file if possible
+    
+When starting a new binary:
+* out-of-space during migrations
+    * Log error in log file if possible
+    * automatic updater will try to rerun the binary on next cycle
+* failure to start
+    * Log error in log file if possible
+    * automatic updater will try to rerun the binary on next cycle
+* not yet configured
+    * Log error in log file if possible
+    * storage node will run a setup or describe out to fix the problem...?
+
+When performing a gradual rollouts.
+* bad gradual rollout. We know it's a bad rollout if our application stops working
+    * Log error in log file if possible
+    * if we did have a database migration, api/grpc change, or file system change in the latest update then wait for next update???
+        * bad latest update might have had faulty database changes that will need to be migrated again.
+    * otherwise rollback to previous version
```

Comment:
What is the whole process for handling bad gradual rollout. How do we get the information that 15% of nodes are not coming up. Who will read the message. Who will flip the switch to turn rollout off. We need the whole process here.